# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/PreallocatedProcessManager.h
# Commit: 2cf5cad90099
# Full Hash: 2cf5cad900992236d596bfb329099742ff15f689
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2024-06-20 04:08:16
# Regressor Bug: 1728331
# File Overlap Count: 1
# Description:
#   Bug 1728331 - Part 4: Make ContentParent KeepAlives explicit with RAII references, r=smaug,dom-worker-reviewers,asuth
#   
#   This is a fairly significant patch, however it would be difficult to break it
#   down into smaller patches:
#   
# ==============================================================================

diff -r d920c2d72d00 -r 2cf5cad90099 dom/ipc/PreallocatedProcessManager.h
--- a/dom/ipc/PreallocatedProcessManager.h	Wed Jun 19 20:14:49 2024 +0000
+++ b/dom/ipc/PreallocatedProcessManager.h	Wed Jun 19 20:14:50 2024 +0000
@@ -8,13 +8,10 @@
 #define mozilla_PreallocatedProcessManager_h
 
 #include "base/basictypes.h"
-#include "mozilla/AlreadyAddRefed.h"
+#include "mozilla/dom/UniqueContentParentKeepAlive.h"
 #include "nsStringFwd.h"
 
 namespace mozilla {
-namespace dom {
-class ContentParent;
-}  // namespace dom
 
 /**
  * This class manages a ContentParent that it starts up ahead of any particular
@@ -32,7 +29,9 @@
 class PreallocatedProcessManagerImpl;
 
 class PreallocatedProcessManager final {
-  typedef mozilla::dom::ContentParent ContentParent;
+  using ContentParent = mozilla::dom::ContentParent;
+  using UniqueContentParentKeepAlive =
+      mozilla::dom::UniqueContentParentKeepAlive;
 
  public:
   static PreallocatedProcessManagerImpl* GetPPMImpl();
@@ -55,7 +54,7 @@
    * If we use a preallocated process, it will schedule the start of
    * another on Idle (AllocateOnIdle()).
    */
-  static already_AddRefed<ContentParent> Take(const nsACString& aRemoteType);
+  static UniqueContentParentKeepAlive Take(const nsACString& aRemoteType);
 
   /**
    * Note that a process was shut down, and should no longer be tracked as a